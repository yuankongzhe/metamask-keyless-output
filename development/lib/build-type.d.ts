import { Infer, Struct } from 'superstruct';

export type Unique<Element extends Struct<any>> = (
  struct: Struct<Infer<Element>[], Infer<Element>>,
  eq?: (a: Infer<Element>, b: Infer<Element>) => boolean,
) => Struct<Infer<Element>[], Infer<Element>>;
