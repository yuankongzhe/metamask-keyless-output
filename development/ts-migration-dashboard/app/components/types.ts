export type BoxRect = {
  moduleId: string;
  top: number;
  left: number;
  width: number;
  height: number;
  centerX: number;
  centerY: number;
};

export type BoxModel = {
  rect: BoxRect;
  dependencyBoxRects: BoxRect[];
  dependentBoxRects: BoxRect[];
};
