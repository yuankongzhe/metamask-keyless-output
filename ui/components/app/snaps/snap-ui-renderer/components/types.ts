import { Component } from '@metamask/snaps-sdk';

export type UIComponentParams<T extends Component> = {
  map: Record<string, number>;
  element: T;
  form?: string;
};

export type UIComponent = {
  element: string;
  props?: Record<string, unknown>;
  children?: UIComponent[] | string;
  key?: string;
};

export type UIComponentFactory<T extends Component> = (
  params: UIComponentParams<T>,
) => UIComponent;
