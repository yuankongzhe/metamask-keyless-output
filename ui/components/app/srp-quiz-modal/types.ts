import { ButtonSize, ButtonVariant } from '../../component-library';

export enum QuizStage {
  introduction = 'introduction',
  questionOne = 'question_one',
  wrongAnswerQuestionOne = 'wrong_answer_question_one',
  rightAnswerQuestionOne = 'right_answer_question_one',
  questionTwo = 'question_two',
  wrongAnswerQuestionTwo = 'wrong_answer_question_two',
  rightAnswerQuestionTwo = 'right_answer_question_two',
}

export interface IQuizInformationProps {
  /**
   * The icon to display in the modal should use <Icon /> component
   */
  icon?: any;
  /**
   * The image to display in the modal
   */
  image?: string;
  /**
   * The text content to go inside of the <Text /> component
   */
  content: string;
  /**
   * More text content to go inside of the <Text /> component
   */
  moreContent?: string;
  /**
   * Array of <Button /> component props
   */
  buttons: {
    onClick: () => void;
    label: string;
    variant: ButtonVariant;
    size?: ButtonSize;
    'data-testid'?: string;
  }[];
}

export type JSXDict = { [key: string]: () => JSX.Element };
