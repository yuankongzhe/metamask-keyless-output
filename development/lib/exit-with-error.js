/**
 * Exit the process with an error message.
 *
 * Note that this should be called before the process ends, but it will not
 * itself end the process. This is because the Node.js documentation strongly
 * advises against calling `process.exit` directly.
 *
 * @param {string} errorMessage - The error message that is causing the non-
 *   zero exit code.
 */
function exitWithError(errorMessage) {
  console.error(errorMessage);
  process.exitCode = 1;
  if (process.env.CI) {
    // For some reason, the CI environment does not exit when process.exitCode
    // is set, or at least doesn't terminate immediately. Something may be
    // keeping the event loop from ending. The result is that sometimes jobs
    // fail because of timeout even though none of the individual tests fail.
    // This is a workaround.
    process.exit();
  }
}

module.exports = { exitWithError };
