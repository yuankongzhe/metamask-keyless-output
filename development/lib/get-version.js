const { version: manifestVersion } = require('../../package.json');
const { loadBuildTypesConfig } = require('./build-type');

/**
 * Get the current version of the MetaMask extension. The base manifest version
 * is modified according to the build type and version.
 *
 * The build version is needed because certain build types (such as beta) may
 * be released multiple times during the release process.
 *
 * @param {string} buildType - The build type.
 * @param {number} buildVersion - The build version.
 * @returns {string} The MetaMask extension version.
 */
function getVersion(buildType, buildVersion) {
  return loadBuildTypesConfig().buildTypes[buildType].isPrerelease === true
    ? `${manifestVersion}-${buildType}.${buildVersion}`
    : manifestVersion;
}

module.exports = { getVersion };
