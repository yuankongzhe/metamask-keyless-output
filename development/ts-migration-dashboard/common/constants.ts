import path from 'path';

export const ROOT_DIRECTORY_PATH = path.resolve(__dirname, '../../..');
export const PROJECT_DIRECTORY_PATH = path.join(
  ROOT_DIRECTORY_PATH,
  'development',
  'ts-migration-dashboard',
);
export const COMMON_DIRECTORY_PATH = path.join(
  PROJECT_DIRECTORY_PATH,
  'common',
);
export const APP_DIRECTORY_PATH = path.join(PROJECT_DIRECTORY_PATH, 'app');
export const INTERMEDIATE_BUILD_DIRECTORY_PATH = path.join(
  PROJECT_DIRECTORY_PATH,
  'build',
  'intermediate',
);
export const FINAL_BUILD_DIRECTORY_PATH = path.join(
  PROJECT_DIRECTORY_PATH,
  'build',
  'final',
);
export const ENTRYPOINT_PATTERNS = [
  'app/scripts/background',
  'app/scripts/contentscript',
  'app/scripts/disable-console',
  'app/scripts/inpage',
  'app/scripts/phishing-detect',
  'app/scripts/sentry-install',
  'app/scripts/ui',
  'development/build/index',
  '**/*.stories',
  '**/*.test',
];
export const FILES_TO_CONVERT_PATH = path.join(
  PROJECT_DIRECTORY_PATH,
  'files-to-convert.json',
);
