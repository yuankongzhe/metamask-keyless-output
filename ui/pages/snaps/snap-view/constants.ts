export const KeyringSnapRemovalResultStatus = {
  Success: 'success',
  Failed: 'failed',
  None: 'none',
};
