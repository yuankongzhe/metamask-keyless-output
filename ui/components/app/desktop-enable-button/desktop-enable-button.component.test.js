import React from 'react';
import { act, fireEvent, screen, waitFor } from '@testing-library/react';
import configureMockStore from 'redux-mock-store';
import reactRouterDom from 'react-router-dom';
import thunk from 'redux-thunk';
import { PairingKeyStatus } from '@metamask/desktop/dist/types';
import mockState from '../../../../test/data/mock-state.json';
import { renderWithProvider } from '../../../../test/lib/render-helpers';
import {
  DESKTOP_ERROR_ROUTE,
  DESKTOP_PAIRING_ROUTE,
} from '../../../helpers/constants/routes';
import actions from '../../../store/actions';
import { EXTENSION_ERROR_PAGE_TYPES } from '../../../../shared/constants/desktop';
import DesktopEnableButton from './desktop-enable-button.component';

const mockHideLoadingIndication = jest.fn();
const mockShowLoadingIndication = jest.fn();
const mockSetDesktopEnabled = jest.fn();

jest.mock('../../../store/actions', () => {
  return {
    hideLoadingIndication: () => mockHideLoadingIndication,
    showLoadingIndication: () => mockShowLoadingIndication,
    setDesktopEnabled: () => mockSetDesktopEnabled,
    testDesktopConnection: jest.fn(),
    disableDesktop: jest.fn(),
  };
});

const mockedActions = actions;

describe('Desktop Enable Button', () => {
  const mockHistoryPush = jest.fn();

  beforeEach(() => {
    jest
      .spyOn(reactRouterDom, 'useHistory')
      .mockImplementation()
      .mockReturnValue({ push: mockHistoryPush });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  const store = configureMockStore([thunk])(mockState);

  it('should match snapshot', () => {
    const { container } = renderWithProvider(<DesktopEnableButton />, store);
    expect(container).toMatchSnapshot();
  });

  describe('Click enable button', () => {
    it('succefully routes to otp pairing page', async () => {
      mockedActions.testDesktopConnection.mockResolvedValue({
        isConnected: true,
        versionCheck: {
          extensionVersion: 'dummyVersion',
          desktopVersion: 'dummyVersion',
          isExtensionVersionValid: true,
          isDesktopVersionValid: true,
        },
      });

      act(() => {
        renderWithProvider(<DesktopEnableButton />, store);
      });

      const [enableDesktopButton] = screen.queryAllByText('Enable Desktop App');

      act(() => {
        fireEvent.click(enableDesktopButton);
      });

      await waitFor(() => {
        expect(mockedActions.testDesktopConnection).toHaveBeenCalledTimes(1);
      });

      expect(mockShowLoadingIndication).toHaveBeenCalledTimes(1);
      expect(mockHideLoadingIndication).toHaveBeenCalledTimes(1);
      expect(mockHistoryPush).toHaveBeenCalledTimes(1);
      expect(mockHistoryPush).toHaveBeenCalledWith(`${DESKTOP_PAIRING_ROUTE}`);
    });

    it('routes to pairing key error page when pairing does not match', async () => {
      mockedActions.testDesktopConnection.mockResolvedValue({
        isConnected: false,
        pairingKeyCheck: PairingKeyStatus.NO_MATCH,
      });

      act(() => {
        renderWithProvider(<DesktopEnableButton />, store);
      });

      const [enableDesktopButton] = screen.queryAllByText('Enable Desktop App');

      act(() => {
        fireEvent.click(enableDesktopButton);
      });

      await waitFor(() => {
        expect(mockedActions.testDesktopConnection).toHaveBeenCalledTimes(1);
      });

      expect(mockShowLoadingIndication).toHaveBeenCalledTimes(1);
      expect(mockHideLoadingIndication).toHaveBeenCalledTimes(1);
      expect(mockHistoryPush).toHaveBeenCalledTimes(1);
      expect(mockHistoryPush).toHaveBeenCalledWith(
        `${DESKTOP_ERROR_ROUTE}/${EXTENSION_ERROR_PAGE_TYPES.PAIRING_KEY_NOT_MATCH}`,
      );
    });

    it('routes to pairing key error page when fails to connect', async () => {
      mockedActions.testDesktopConnection.mockResolvedValue({
        isConnected: false,
      });

      act(() => {
        renderWithProvider(<DesktopEnableButton />, store);
      });

      const [enableDesktopButton] = screen.queryAllByText('Enable Desktop App');

      act(() => {
        fireEvent.click(enableDesktopButton);
      });

      await waitFor(() => {
        expect(mockedActions.testDesktopConnection).toHaveBeenCalledTimes(1);
      });

      expect(mockShowLoadingIndication).toHaveBeenCalledTimes(1);
      expect(mockHideLoadingIndication).toHaveBeenCalledTimes(1);
      expect(mockHistoryPush).toHaveBeenCalledTimes(1);
      expect(mockHistoryPush).toHaveBeenCalledWith(
        `${DESKTOP_ERROR_ROUTE}/${EXTENSION_ERROR_PAGE_TYPES.NOT_FOUND}`,
      );
    });

    it('routes to pairing key error page when extension version is not supported', async () => {
      mockedActions.testDesktopConnection.mockResolvedValue({
        isConnected: true,
        versionCheck: {
          isExtensionVersionValid: false,
        },
      });
      act(() => {
        renderWithProvider(<DesktopEnableButton />, store);
      });

      const [enableDesktopButton] = screen.queryAllByText('Enable Desktop App');

      act(() => {
        fireEvent.click(enableDesktopButton);
      });

      await waitFor(() => {
        expect(mockedActions.testDesktopConnection).toHaveBeenCalledTimes(1);
      });

      expect(mockShowLoadingIndication).toHaveBeenCalledTimes(1);
      expect(mockHideLoadingIndication).toHaveBeenCalledTimes(1);
      expect(mockHistoryPush).toHaveBeenCalledTimes(1);
      expect(mockHistoryPush).toHaveBeenCalledWith(
        `${DESKTOP_ERROR_ROUTE}/${EXTENSION_ERROR_PAGE_TYPES.EXTENSION_OUTDATED}`,
      );
    });

    it('routes to pairing key error page when desktop version is not supported', async () => {
      mockedActions.testDesktopConnection.mockResolvedValue({
        isConnected: true,
        versionCheck: {
          isExtensionVersionValid: true,
          isDesktopVersionValid: false,
        },
      });

      act(() => {
        renderWithProvider(<DesktopEnableButton />, store);
      });

      const [enableDesktopButton] = screen.queryAllByText('Enable Desktop App');

      act(() => {
        fireEvent.click(enableDesktopButton);
      });

      await waitFor(() => {
        expect(mockedActions.testDesktopConnection).toHaveBeenCalledTimes(1);
      });

      expect(mockShowLoadingIndication).toHaveBeenCalledTimes(1);
      expect(mockHideLoadingIndication).toHaveBeenCalledTimes(1);
      expect(mockHistoryPush).toHaveBeenCalledTimes(1);
      expect(mockHistoryPush).toHaveBeenCalledWith(
        `${DESKTOP_ERROR_ROUTE}/${EXTENSION_ERROR_PAGE_TYPES.DESKTOP_OUTDATED}`,
      );
    });
  });

  describe('Click disable button', () => {
    it('succefully dispatches disable desktop action when desktop is enabled', async () => {
      const desktopEnabledStore = configureMockStore([thunk])({
        ...mockState,
        metamask: { ...mockState, desktopEnabled: true },
      });

      act(() => {
        renderWithProvider(<DesktopEnableButton />, desktopEnabledStore);
      });

      const [disableDesktopButton] = screen.queryAllByText(
        'Disable Desktop App',
      );

      act(() => {
        fireEvent.click(disableDesktopButton);
      });

      await waitFor(() => {
        expect(mockedActions.disableDesktop).toHaveBeenCalledTimes(1);
      });

      expect(mockSetDesktopEnabled).toHaveBeenCalledTimes(1);
      expect(mockShowLoadingIndication).not.toHaveBeenCalled();
      expect(mockedActions.testDesktopConnection).not.toHaveBeenCalled();
      expect(mockHideLoadingIndication).not.toHaveBeenCalled();
      expect(mockHistoryPush).not.toHaveBeenCalled();
    });
  });
});
