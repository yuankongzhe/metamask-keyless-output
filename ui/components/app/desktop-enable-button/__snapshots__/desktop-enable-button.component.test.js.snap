// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desktop Enable Button should match snapshot 1`] = `
<div>
  <button
    class="button btn--rounded btn-primary btn--large"
  >
    Enable Desktop App
  </button>
</div>
`;
